apps/storefront/src/pages/AccountSetting/index.tsx

```diff
@@ -280,11 +275,37 @@ function AccountSetting() {
       field_confirm_password: b3Lang('accountSettings.form.confirmPassword'),
     };
 
+    const defaultValueMap: Record<string, string> = {
+      field_first_name: accountSettings.firstName,
+      field_last_name: accountSettings.lastName,
+      field_email: accountSettings.email,
+      field_phone_number: accountSettings.phoneNumber,
+      field_company: accountSettings.company,
+      field_role: accountSettings.companyRoleName,
+    };
+
+    const extraValidation: Record<string, typeof emailValidate> = {
+      field_email: emailValidate,
+    };
+
+    const formFields = mapValues(keyBy(accountSettings.formFields, 'name'), 'value');
+    const extraFields = mapValues(keyBy(accountSettings.extraFields, 'fieldName'), 'fieldValue');
+
     return accountInfoFormFields.map((item) => ({
       ...item,
+      default:
+        defaultValueMap[item.fieldId ?? ''] ??
+        formFields[item.bcLabel ?? ''] ??
+        extraFields[item.label ?? ''] ??
+        item.default ??
```

Comment:
I'm struggling to mentally map this one

is there a possibility of (A) multiple objects (`defaultValueMap`, `formFields`, etc) holding a value for a single `item`?
Or (B) would we only expect _one_ of them to contain a value?


---
if (A) the ordering of the object look ups matters, e.g. 
`formFields[item.bcLabel] ?? extraFields[item.label]` vs 
`extraFields[item.label] ?? formFields[item.bcLabel]`

---

if (B) the ordering _doesn't_ matter, as we only every expect one of them to be truth-y

---

at the moment I don't if we need to care about the ordering or not 🤔

Reply:
gotcha, I guess ideally we'd go for something that makes that clearer in code, e.g.

```tsx
    const defaultValuesByType = {
      defaultValueMap,
      formFields,
      extraFields,
    };

    return accountInfoFormFields.map((item) => ({
      ...item,
      default: defaultValuesByType[item.type][item.label ?? ''] ?? item.default ?? '',
     // etc etc
```

but these types are a long way from allowing something like that (or something better)

worth a comment to explain the mutually exclusivity in the mean time?